### PR TITLE
fix(route):  NASA 每日天文图片

### PR DESCRIPTION
**Pull request by williamgateszhao**
## 该 PR 相关 Issue / Involved Issue

Close #

## 路由地址示例 / Example for the Proposed Route(s)

```routes
/nasa/apod
/nasa/apod-ncku
/nasa/apod-cn
```

## 新 RSS 路由检查表 / New RSS Route Checklist
  
- [ ] 新的路由 New Route
  - [x] 跟随 [v2 路由规范](https://docs.rsshub.app/joinus/script-standard.html) Follows [v2 Script Standard](https://docs.rsshub.app/en/joinus/script-standard.html)
- [ ] 文档说明 Documentation
  - [ ] 中文文档 CN
  - [ ] 英文文档 EN
- [x] 全文获取 fulltext
  - [x] 使用缓存 Use Cache
- [ ] 反爬/频率限制 anti-bot or rate limit?
  - [ ] 如果有, 是否有对应的措施? If yes, do your code reflect this sign?
- [x] [日期和时间](https://docs.rsshub.app/joinus/pub-date.html) [date and time](https://docs.rsshub.app/en/joinus/pub-date.html)
  - [x] 可以解析 Parsed
  - [x] 时区调整 Correct TimeZone
- [ ] 添加了新的包 New package added
- [ ] `Puppeteer`

## 说明 / Note

将该路由转换为v2路由，并主要修复了两个问题：
原 `/nasa/apod-cn` 路由已损坏，修复到可用状态
原 `/nasa/apod` 和 `/nasa/apod-ncku` 路由只能输出当天图片，为防止rss阅读器漏取内容，现修改为输出10条内容

暂无法解决的问题：
暂不了解如何在使用 got 对外请求时选择 ip family，而 nasa 的域名 `apod.nasa.gov` 在本 pr 提交时 ipv6 是坏的，对于自建rsshub来说，如果服务器有ipv6地址，可能无法使用 `/apod` 这个路由

### CODE REVIEW

File: lib/v2/nasa/apod-cn.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
+const got = require('@/utils/got');
+const cheerio = require('cheerio');
+const { parseDate } = require('@/utils/parse-date');
+
+module.exports = async (ctx) => {
+    const rootUrl = `https://www.nasachina.cn/astronomy-picture-of-the-day`;
```
Comment (TonyRL) [inline lib/v2/nasa/apod-cn.js:6]: This site has a JSON API. Please use that instead.
`/wp-json/wp/v2/categories` will list all the categories. The category id for apod is `2`.
`/wp-json/wp/v2/posts` will list all the posts. You can filter the posts by using `categories` search params and `per_page` if you want more than 10 posts.

File: lib/v2/nasa/apod-cn.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+            ctx.cache.tryGet(item.link, async () => {
+                const detailResponse = await got({
+                    method: 'get',
+                    url: item.link,
+                });
+                const content = cheerio.load(detailResponse.data);
+
+                const description = content('main article').html();
+                const dateText = content('time.updated').attr('datetime');
+                const pubDate = dateText ? parseDate(dateText) : new Date();
```
Comment (TonyRL) [inline lib/v2/nasa/apod-cn.js:35]: Don't add a pubDate if the site does not provide one.
docs: https://docs.rsshub.app/joinus/pub-date.html

File: lib/v2/nasa/apod-ncku.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    const rootUrl = 'http://sprite.phys.ncku.edu.tw/astrolab/mirrors/apod/archivepix.html';
+    const response = await got({
+        method: 'get',
+        url: rootUrl,
+    });
+
+    const $ = cheerio.load(response.data);
+
+    const list = $('body > b > a')
+        .slice(0, 10)
```
Comment (TonyRL) [inline lib/v2/nasa/apod-ncku.js:15]: You may pass the `ctx.query.limit` common parameter here which allows users to adjust no. to items to fetch
related user docs: https://docs.rsshub.app/parameter.html

File: lib/v2/nasa/apod.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+module.exports = async (ctx) => {
+    const rootUrl = 'https://apod.nasa.gov/apod/archivepix.html';
+    const response = await got({
+        method: 'get',
+        url: rootUrl,
+    });
+    const $ = cheerio.load(response.data);
+
+    const list = $('body > b > a')
+        .slice(0, 10)
```
Comment (TonyRL) [inline lib/v2/nasa/apod.js:15]: You may pass the `ctx.query.limit` common parameter here which allows users to adjust no. to items to fetch
related user docs: https://docs.rsshub.app/parameter.html

File: lib/v2/nasa/apod-cn.js (hunk 2 above)
Comment (williamgateszhao) [reply lib/v2/nasa/apod-cn.js:35]: 这是 "Astronomy Picture of the Day" ，网站在页面上、链接中均提供了日期，根据上述规范，应该提供pubDate

File: lib/v2/nasa/apod-cn.js [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+        method: 'get',
+        url: rootUrl,
+    });
+
+    const items = await Promise.all(
+        data.map((item) =>
+            ctx.cache.tryGet(item.link, async () => {
+                const detailResponse = await got({
+                    method: 'get',
+                    url: item.link,
```
Comment (TonyRL) [inline lib/v2/nasa/apod-cn.js:18]: No need to visit `item.link`. The full article is already included in the /posts request(content.rendered).

**Pull request merged**